### PR TITLE
Bugs/1020 support index function

### PR DESCRIPTION
**Pull request by NaKister**
Fixed the bug and did complete support for the INDEX Function.

### CODE REVIEW

File: EPPlusTest/FormulaParsing/Excel/Functions/RefAndLookup/IndexTests.cs [hunk 1]
```diff
@@ -98,6 +89,686 @@ public void IndexHandlesNAError()
...
+			this.Worksheet.Cells["B2"].Value = 6;
+			this.Worksheet.Cells["B3"].Value = 4;
+			this.Worksheet.Cells["B4"].Value = 9;
+			this.Worksheet.Cells["C2"].Formula = "INDEX(B1:B4, , )";
+			this.Worksheet.Calculate();
+			Assert.AreEqual(eErrorType.Value, ((ExcelErrorValue)this.Worksheet.Cells["C2"].Value).Type);
+		}
+
+		[TestMethod]
+		public void IndexWithRowAsRefereneToCellReturnsCorrectValue()
```
Comment (emdelaney) [inline EPPlusTest/FormulaParsing/Excel/Functions/RefAndLookup/IndexTests.cs]: *Reference

File: EPPlus/FormulaParsing/Excel/Functions/RefAndLookup/Index.cs [hunk 2]
```diff
@@ -27,38 +27,95 @@
...
+			var args = cellRange.Value as IEnumerable<FunctionArgument>;
+			var result = new CompileResultFactory();
 			if (args != null)
 			{
 				var index = this.ArgToInt(arguments, 1);
 				if (index > args.Count())
 					throw new ExcelErrorValueException(eErrorType.Ref);
 				var candidate = args.ElementAt(index - 1);
-				return crf.Create(candidate.Value);
+				return result.Create(candidate.Value);
```
Comment (emdelaney) [inline EPPlus/FormulaParsing/Excel/Functions/RefAndLookup/Index.cs]: Isn't there like a base.CreateResult method that basically does this?

File: EPPlus/FormulaParsing/Excel/Functions/RefAndLookup/Index.cs [hunk 3]
```diff
@@ -27,38 +27,95 @@
...
+				if ((column == 0 && row == 0) || column < 0)
+					return new CompileResult(eErrorType.Value);
+
+				var rangeInfo = cellRange.ValueAsRangeInfo;
+				if (rangeInfo.Address.Rows == 1 && arguments.Count() < 3)
+				{
+					column = row;
+					row = 1;
+				}
+
```
Comment (emdelaney) [inline EPPlus/FormulaParsing/Excel/Functions/RefAndLookup/Index.cs]: All of these extra newlines can go away :)

File: EPPlus/FormulaParsing/Excel/Functions/RefAndLookup/Index.cs (hunk 2 above)
Comment (NaKister) [reply EPPlus/FormulaParsing/Excel/Functions/RefAndLookup/Index.cs]: Yes there is. I'm changing all of the calls to result to be of this base.CreateResult call.

File: EPPlusTest/FormulaParsing/Excel/Functions/RefAndLookup/IndexTests.cs (hunk 1 above)
Comment (NaKister) [reply EPPlusTest/FormulaParsing/Excel/Functions/RefAndLookup/IndexTests.cs]: Got it :)

File: EPPlus/FormulaParsing/Excel/Functions/RefAndLookup/Index.cs (hunk 3 above)
Comment (NaKister) [reply EPPlus/FormulaParsing/Excel/Functions/RefAndLookup/Index.cs]: Got it :)

**Pull request merged**